cleanup (i guess?) & content - med cat and med cat app thoughts

arisaema12 opened the pull request:
<!-- Write BELOW The Headers and ABOVE The comments else it may not be viewable. -->
<!-- You can view CONTRIBUTING.md for a detailed description of the pull request process. -->
<!-- Be sure to name your PR something descriptive and succinct; include Bugfix: Feature: Enhancement: or Content: in the title to describe what type of PR it is. -->

## About The Pull Request
was originally just pronoun-ifying, but got carried away and ended up adding loads of random status constraints to the med cat file and new thoughts to both med cat and med cat app... oops
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage senior developers from merging your PR! -->

## Why This Is Good For ClanGen
med cats can no longer generate thoughts about kits/leader/deputy when there aren't any in the clan. new thoughts add variety to clangen.

plus scribble seemed like she needed more prs to look through :D /j
<!-- If this is a bug fix, you can remove this section. -->
<!-- Please add a short description of why you think these changes would benefit the game. -->
<!-- If this PR was the result of discussion/testing on the discord, please add a link to the discord conversation here. -->
<!-- If you have multiple features that can stand on their own, or unrelated bugfixes, please create separate PRs for them. -->

## Proof of Testing
<img width="392" alt="charismatic_cat_new_thought" src="https://github.com/user-attachments/assets/cf4ced51-b26d-4941-8a46-cd381b9b957a">

<!-- Include any screenshots, debugging steps, or links to beta testing threads here. At least one form of proof of testing is REQUIRED for all new content. You must be able to run the code locally before you PR it here. -->

## Changelog/Credits

- medicine cats will no longer generate thoughts about kits/leader/deputy when there aren't any in the clan
- added new medicine cat and medicine cat apprentice thoughts

arisaema12 committed:
Update medicine_cat.json

arisaema12 committed:
Update medicine_cat_apprentice.json

selkirks commented on resources/dicts/thoughts/alive/medicine_cat.json:
```diff
@@ -73,6 +118,18 @@
+    {
+        "id": "calm_medicine_cat_apprentice",
+        "thoughts": [
+            "Calms down an apprentice who is panicking other their minor injury"
```
shld be 'over their minor injury'

scribblecrumb replied:
"their" also needs to be pronoun tagged

arisaema12 replied:
ah I thought I should leave it ambiguous as to which apprentice the thought refers to, but i'll change it to be pronoun tagged

selkirks commented on resources/dicts/thoughts/alive/medicine_cat.json:
```diff
@@ -232,6 +300,15 @@
+    {
+      "id": "playful_medicine_cat",
+        "thoughts": [
+            "Keeps {PRONOUN/m_c/poss} patients distracted with silly games while {PRONOUN/m_c/subject} treat them"
```
treat shld be verb tagged

scribblecrumb commented on resources/dicts/thoughts/alive/medicine_cat.json:
```diff
@@ -86,7 +143,8 @@
         "thoughts": [
-            "Is doing {PRONOUN/m_c/poss} daily checkup on the elders"
+            "Is doing {PRONOUN/m_c/poss} daily checkup on the elders",
+	    "Easily persuades another Clan's medicine cat to gift c_n herbs"
```
We have moon events that mirror this and are also capable of actually changing the clan's herb supply... perhaps we could reword this a bit? Perhaps into "Is considering the best way to ask another Clan for herbs"?  or "Is coming up with just the right words to ask another Clan's medicine cat for extra herbs"

scribblecrumb commented on resources/dicts/thoughts/alive/medicine_cat_apprentice.json:
```diff
@@ -273,5 +273,32 @@
+        "thoughts": [
+            "Thinks r_c is far too soft with {PRONOUN/r_c/poss} patients",
+	    "Wishes r_c wasn't so soft",
+	    "Has to promice r_c {PRONOUN/m_c/subject} won't gather deathberries again"
```
```suggestion
	    "Has to promise r_c {PRONOUN/m_c/subject} won't gather deathberries again"
```

arisaema12 committed:
spelling

someone remind me to use a spelling-checker next time i try and write anything XD

Co-authored-by: scribblecrumb <115410010+scribblecrumb@users.noreply.github.com>

arisaema12 committed:
Update medicine_cat.json
